kmapper/visuals.py

```diff
@@ -476,20 +516,69 @@ def _format_tooltip(
         nbins,
     )
 
-    tooltip = env.get_template("cluster_tooltip.html").render(
-        projection_stats=projection_stats,
-        cluster_stats=cluster_stats,
-        custom_tooltips=custom_tooltips,
+    tooltip_data = {
+        'projection_stats': projection_stats,
+        'cluster_stats': cluster_stats,
+        'custom_tooltips': custom_tooltips,
+        'histogram': histogram,
+        'dist_label': "Member",
+        'node_id': node_ID,
+    }
+
+    return tooltip_data
+
+def render_d3_vis(
+    title,
+    mapper_summary,
+    histogram,
+    mapper_data,
+    colorscale
+    ):
+    # Find the module absolute path and locate templates
+    module_root = os.path.join(os.path.dirname(__file__), "templates")
+    env = Environment(loader=FileSystemLoader(module_root))
+
+    # Find the absolute module path and the static files
+    js_path = os.path.join(os.path.dirname(__file__), "static", "kmapper.js")
+    with open(js_path, "r") as f:
+        js_text = f.read()
+
+    css_path = os.path.join(os.path.dirname(__file__), "static", "style.css")
+    with open(css_path, "r") as f:
+        css_text = f.read()
+
+    if np.array(histogram).ndim == 1:
+        histogram = [histogram]
+
+    color_function_names = mapper_summary['custom_meta'].pop('color_function', None)
+    if isinstance(color_function_names, str):
+        color_function_names = [color_function_name]
+
+    if color_function_names is None:
+        raise Exception('''`color_function` required but not found in mapper_summary `custom_meta`''')
+
+    number_of_histograms = np.array(histogram).shape[0]
+    number_of_color_function_names = np.array(color_function_names).shape[0]
+    if number_of_histograms != number_of_color_function_names:
+        raise Exception('Number of histograms ({}) does not equal number of color_function_names ({})'.format(number_of_histograms, number_of_color_function_names))
+
+    # Render the Jinja template, filling fields as appropriate
+    html = env.get_template("base.html").render(
+        title=title,
+        mapper_summary=mapper_summary,
+        color_function_names=color_function_names,
         histogram=histogram,
-        dist_label="Member",
-        node_id=node_ID,
+        dist_label="Node",
+        mapper_data=mapper_data,
+        colorscale=colorscale,
+        js_text=js_text,
+        css_text=css_text,
     )
 
-    return tooltip
-
+    return html
 
 def _node_color_function(member_ids, color_values):
-    return np.mean(color_values[member_ids])
+    return np.mean(color_values[member_ids], axis=0)
```

Comment:
I would *love* it if the user could supply this function.

Reply:
Yeah me too -- I forgot to come back and do this, got distracted with the html/js. We'll have to document it carefully, because with the possibility to pass multiple color_value columns now, the function passed here needs to always do axis=0 to preserve the mapping to the color_value shape. Maybe we... add axis=0 to any function they provide, if it's missing? 🤷 Too opinionated? I think the code would crash if not.

Reply:
What about just taking a string argument that is `mean`, `median`, `max`,  or `min` for now?  It's not a complete solution, but should be pretty easy to implement.  I assume that opening up that option will in turn produce requests for different functions also.  When and if that happens we can consider a more general pattern.

Reply:
That's a really good idea, will do

Reply:
done!

Reply:
The way I implemented it, any function available on the `np` base class can be used, specified as a string. I just extract it via `getattr(np, node_function_name)`

Reply:
See https://github.com/scikit-tda/kepler-mapper/pull/196/commits/67fcd69ee0fcc70c63fb5e30f9804e6b30d4f552